x-pack/platform/plugins/shared/ml/public/application/components/controls/select_severity/select_severity.tsx

```diff
@@ -9,154 +9,253 @@
+import { EuiHealth } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
-import { FormattedMessage } from '@kbn/i18n-react';
 import { usePageUrlState } from '@kbn/ml-url-state';
-import { getSeverityColor, ML_ANOMALY_THRESHOLD } from '@kbn/ml-anomaly-utils';
-
-const warningLabel: string = i18n.translate('xpack.ml.controls.selectSeverity.warningLabel', {
-  defaultMessage: 'warning',
-});
-const minorLabel: string = i18n.translate('xpack.ml.controls.selectSeverity.minorLabel', {
-  defaultMessage: 'minor',
-});
-const majorLabel: string = i18n.translate('xpack.ml.controls.selectSeverity.majorLabel', {
-  defaultMessage: 'major',
-});
-const criticalLabel: string = i18n.translate('xpack.ml.controls.selectSeverity.criticalLabel', {
-  defaultMessage: 'critical',
-});
-
-const optionsMap = {
-  [warningLabel]: ML_ANOMALY_THRESHOLD.LOW,
-  [minorLabel]: ML_ANOMALY_THRESHOLD.MINOR,
-  [majorLabel]: ML_ANOMALY_THRESHOLD.MAJOR,
-  [criticalLabel]: ML_ANOMALY_THRESHOLD.CRITICAL,
-};
+import { ML_ANOMALY_THRESHOLD } from '@kbn/ml-anomaly-utils';
+import type { SeverityThreshold } from '../../../../../common/types/anomalies';
+import { MultiSuperSelect } from '../../multi_super_select/multi_super_select';
+import { useSeverityOptions } from '../../../explorer/hooks/use_severity_options';
 
 export interface TableSeverityPageUrlState {
   pageKey: 'mlSelectSeverity';
-  pageUrlState: TableSeverity;
+  pageUrlState: TableSeverityState;
 }
 
 export interface TableSeverity {
   val: number;
   display: string;
   color: string;
+  threshold: SeverityThreshold;
 }
 
-export const SEVERITY_OPTIONS: TableSeverity[] = [
-  {
-    val: ML_ANOMALY_THRESHOLD.LOW,
-    display: warningLabel,
-    color: getSeverityColor(ML_ANOMALY_THRESHOLD.LOW),
-  },
-  {
-    val: ML_ANOMALY_THRESHOLD.MINOR,
-    display: minorLabel,
-    color: getSeverityColor(ML_ANOMALY_THRESHOLD.MINOR),
-  },
-  {
-    val: ML_ANOMALY_THRESHOLD.MAJOR,
-    display: majorLabel,
-    color: getSeverityColor(ML_ANOMALY_THRESHOLD.MAJOR),
-  },
-  {
-    val: ML_ANOMALY_THRESHOLD.CRITICAL,
-    display: criticalLabel,
-    color: getSeverityColor(ML_ANOMALY_THRESHOLD.CRITICAL),
-  },
-];
-
-export function optionValueToThreshold(value: number) {
-  // Get corresponding threshold object with required display and val properties from the specified value.
-  let threshold = SEVERITY_OPTIONS.find((opt) => opt.val === value);
-
-  // Default to warning if supplied value doesn't map to one of the options.
-  if (threshold === undefined) {
-    threshold = SEVERITY_OPTIONS[0];
-  }
-
-  return threshold;
+export interface TableSeverityState {
+  val: SeverityThreshold[];
 }
 
-const TABLE_SEVERITY_DEFAULT = SEVERITY_OPTIONS[0];
+/**
+ * React hook that returns the default table severity state
+ */
+const useTableSeverityDefault = () => {
+  const severityOptions = useSeverityOptions();
+  return useMemo(
+    () => ({
+      val: [severityOptions[0].threshold],
```

Comment:
We should default to showing all severities, which would be consistent with the current behavior.